sklearn/tests/test_impute.py

```diff
@@ -84,12 +85,15 @@ def test_imputation_shape():
     X = np.random.randn(10, 2)
     X[::2] = np.nan
 
-    for strategy in ['mean', 'median', 'most_frequent']:
-        imputer = SimpleImputer(strategy=strategy)
+    for strategy in ['mean', 'median', 'most_frequent', 'mice']:
```

Comment:
I would do a different test:

```python
@pytest.mark.parametrize(
    "imputer",
    [(SimpleImputer()),
     (MICEImputer())]
)
def test_imputation_shape(imputer):
    for strategy in ['mean', 'median', 'most_frequent']:
         # either check if imputer as init_strategy or startegy to set the parameter and apply the remaining test
```

Reply:
I came up with another clean way to do it that I'll use if you don't mind:

```
    # Verify the shapes of the imputed matrix for different strategies.
    X = np.random.randn(10, 2)
    X[::2] = np.nan

    for strategy in ['mean', 'median', 'most_frequent']:
        imputer = SimpleImputer(strategy=strategy)
        mice_imputer = MICEImputer(initial_strategy=strategy)
        X_imputed = imputer.fit_transform(sparse.csr_matrix(X))
        assert X_imputed.shape == (10, 2)
        X_imputed = imputer.fit_transform(X)
        assert X_imputed.shape == (10, 2)
        X_imputed = mice_imputer.fit_transform(X)
        assert X_imputed.shape == (10, 2)
```